alphabot/memory.py

```diff
@@ -59,9 +60,15 @@ def __init__(self):
 
     @gen.coroutine
     def _save(self, key, value):
-        self.r.set(key, value)
+        json_data = json.dumps(value)
+        self.r.set(key, json_data)
 
     @gen.coroutine
     def _get(self, key, default=None):
-        value = self.r.get(key) or default
-        raise gen.Return(value)
+        raw_data = self.r.get(key) or default
+        try:
+            json_data = json.loads(raw_data)
+            raise gen.Return(json_data)
+        except ArithmeticError as e:
```

Comment:
I recommend doing a blanked catch-all error since you are logging the error below.